zenoh-flow/src/utils.rs

```diff
@@ -12,34 +12,60 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
+use crate::model::{Middleware, ZFUri};
 use crate::prelude::ErrorKind;
 use crate::{bail, zferror, Result};
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 use url::Url;
 
 /// Given a string representing a [`Url`](`url::Url`), transform it into a
-/// [`PathBuf`](`std::path::PathBuf`).
+/// [`URIStruct`](`zenoh_flow::runtime:dataflow::loader::URIStruct`).
```

Comment:
```suggestion
/// [`ZFUri`](`zenoh_flow::model::ZFUri`).
```

(not sure about the path)